### PR TITLE
github: Fetch parent repo information from API

### PR DESCRIPTION
**Pull request by indradhanush**
Part of #54741. Forked (pun intended) out from #54663.

**A forked repo on a fresh instance after code host sync, contains parent info**

image

image

**A parent repo on a fresh instance after code host sync, does not contain parent info**

image

image

## Test plan

- Added and updated tests
- Tested locally that coded host sync updates the metadata for GitHub.com and GHE repos

### CODE REVIEW

File: internal/extsvc/github/v4.go [hunk 1]
```diff
@@ -565,7 +565,7 @@ func (c *V4Client) buildGetReposBatchQuery(ctx context.Context, namesWithOwners
 			return "", err
 		}
 		fmt.Fprintf(&b, "repo%d: repository(owner: %q, name: %q) { ", i, owner, name)
-		b.WriteString("... on Repository { ...RepositoryFields } }\n")
+		b.WriteString("... on Repository { ...RepositoryFields parent { nameWithOwner, isFork } } }\n")
```
Comment (eseliger) [inline internal/extsvc/github/v4.go:568]: databaseID can't hurt I think, that way we can make sure 2 repos are actually the same.

File: go.sum
Comment (eseliger) [inline go.sum:1]: how did that happen? 🤔

**Review by eseliger — APPROVED** [review]

File: go.sum
Comment (indradhanush) [reply go.sum:1]: `sg generate go` decided it wants to. 🤷

File: internal/extsvc/github/v4.go (hunk 1 above)
Comment (indradhanush) [reply internal/extsvc/github/v4.go:568]: But will we verify that? And do we need to? We use the `repo.URI` to identify a repo and not the `DatabaseID` so it feels redundant.

**Comment by indradhanush** [comment]
@eseliger Merging. Will follow up if you think databaseID is needed.

**Pull request merged**